contracts/paratii/TcrPlaceholder.sol

```diff
@@ -0,0 +1,169 @@
+pragma solidity ^0.4.18;
+
+import "./ParatiiToken.sol";
+
+/// WARNING , NOT PRODUCTION READY, based on ADCHAIN Registry.
+/// THIS IS JUST A PLACEHOLDER..
+contract TcrPlaceholder {
+
+  // Events
+  // ---------
+
+  event _Application(string videoId, uint deposit);
+  event _NewVideoWhitelisted(string videoId);
+
+  struct Listing {
+    uint applicationExpiry; // Expiration date of apply stage
+    bool whitelisted;       // registration status
+    address owner;          // Owner of the video.
+    uint unstakedDeposit;   // unlocked tokens
+    uint challengeID;       // ID of canonical challenge
+  }
+
+  // Maps of videoId Hashes to listing data.
+  mapping(bytes32 => Listing) public listings;
+
+  ParatiiToken public token;
+  uint public minDeposit;
+  uint public applyStageLen;
+
+  // ---------------------------------------------------------------------------
+  // CONSTRUCTOR
+  // ------------------
+
+  function TcrPlaceholder(
+    address _tokenAddr,
+    uint _minDeposit,
+    uint _applyStageLen
+  ) {
+    token = ParatiiToken(_tokenAddr);
```

Comment:
we have a sort of pattern we use in most of our other contract in which the contract takes the `registryAddress` in the constructor, and then we can get access the other contracts using `Registry(paratiiRegistry).getContract('ParatiiToken')`

Let's do that here as well